### ISSUE TITLE
YOLOv5 loss function

### ISSUE THREAD

**Issue by phuong-2201** [ultralytics/yolov5#7851]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Just a bit curious here, I try to train the model with focal loss (setting fl_gamma = x); the result is really low when compared to the original model. What could be the reason here? I researched and found that objectness suffers from an extreme imbalance between foreground and background class. In my case, I try to detect fish. The fish is tiny. It does not work well with Focal Loss. I hope to receive some insights here to understand the loss function better.

**Comment by thanhnt-3728** [comment]
@phuong-2201 According to this [paper](https://arxiv.org/pdf/1909.04868.pdf), you might also need to change the loss scaling factor and bias initialization in the last convolutional layer.
